### PR TITLE
[#194] Authentication tests using Cypress

### LINKED ISSUE #194: Firebase authentication - signup, login, add/remove user
Test auth flows for signing up and logging in. A user needs to be added in the database and in Firebase, and the token persisted in cookies. 

### Additional context

Before we're able to use `cypress-firebase`, we'll need to manually sign up so we can authenticate using Firebase. We can create a test user and delete the test user after the run, using Firebase admin.

### PR DESCRIPTION
**Pull request by dallashuggins**
### Changes
- Cypress configuration updates
- [RTL support](https://www.npmjs.com/package/@testing-library/cypress) in Cypress tests
- Cypress commands that execute scripts for creating, removing, and updating a user in Firebase and the database
- Tests for auth flows:
  - Logging in as an existing user
  - Signing out when logged in
  - Creating a new account
- New API endpoint for removing a user

Closes #194 

### Testing
- Run `npm run e2e` to start up the application
- The Cypress window will appear eventually
- Click to Run all tests
- Observe test suite run

### Next up
- [ ] #193 
- [ ] #198

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -119,5 +119,13 @@ module.exports = {
...
       },
     },
+    {
+      files: ['**/?(*.)+(spec).[jt]s?(x)'],
+      excludedFiles: ['**/cypress/**/*.[jt]s?(x)'],
+      extends: [...defaultExtends, 'plugin:cypress/recommended'],
+      rules: {
+        ...defaultRules,
+      },
+    },
```
Comment (dallashuggins) [inline .eslintrc.js:129]: Invalid jest errors were being displayed when writing the Cypress tests. Installed the cypress eslint rules and applying to all cypress specific tests now.

File: apps/web-e2e/.env.example [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# API details
+API_URL="http://localhost:3001"
+API_KEY="123"
+
+# Firebase
+FIREBASE_SERVICE_ACCOUNT={"projectId":"","clientEmail":"","privateKey":""}
```
Comment (dallashuggins) [inline apps/web-e2e/.env.example:6]: Requires Firebase admin account to be setup, so a user can be added/removed during testing.

File: apps/web-e2e/src/fixtures/firebase.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "displayName": "Test User",
+  "email": "test@rocketinsights.com",
+  "passphrase": "111111",
+  "password": "Testing123!",
+  "username": "web-e2e-test-user"
+}
```
Comment (dallashuggins) [inline apps/web-e2e/src/fixtures/firebase.json:7]: Details used for creating a test account in Firebase and the database

**Review by smonn — APPROVED** [review]
Feel free to merge this, it works great!

**Review by jakerainis — APPROVED** [review]
👍 👍 👍

File: apps/web-e2e/src/support/commands.ts [hunk 4]
```diff
@@ -1,33 +1,52 @@
...
+/*
+ *
+ * Custom command for initiating the creation of a user
+ * @example cy.createUser()
+ *
+ */
+Cypress.Commands.add(
+  'createUser',
+  (email: string, passphrase: string, password: string, username: string) => {
+    return cy.exec('node ./src/utils/createUser.js', {
```
Comment (afraser) [inline apps/web-e2e/src/support/commands.ts:24]: ah cool! I haven't seen `cy.exec` before

**Review by afraser — APPROVED** [review]
awesommmmme

**Pull request merged**